gtsfm/frontend/correspondence_generator/synthetic_correspondence_generator.py

```diff
@@ -0,0 +1,105 @@
+    KeypointAggregatorDedup,
+)
+from gtsfm.frontend.correspondence_generator.keypoint_aggregator.keypoint_aggregator_unique import (
+    KeypointAggregatorUnique,
+)
+from gtsfm.loader.loader_base import LoaderBase
+from gtsfm.loader.tanks_and_temples_loader import TanksAndTemplesLoader
+
+
+class SyntheticCorrespondenceGenerator(CorrespondenceGeneratorBase):
+    """Pair-wise synthetic keypoint correspondence generator."""
+
+    def __init__(self, dataset_root: str, scene_name: str, deduplicate: bool = True) -> None:
+        """
+        Args:
+            dataset_root: str
+            scene_name: Name of scene from Tanks & Temples dataset.
+            deduplicate: Whether to de-duplicate with a single image the detections received from each image pair.
+        """
+        self._dataset_root = dataset_root
+        self._scene_name = scene_name
+        self._aggregator: KeypointAggregatorBase = (
+            KeypointAggregatorDedup() if deduplicate else KeypointAggregatorUnique()
+        )
+
+    def generate_correspondences(
+        self,
+        client: Client,
+        images: List[Future],
+        image_pairs: List[Tuple[int, int]],
+    ) -> Tuple[List[Keypoints], Dict[Tuple[int, int], np.ndarray]]:
+        """Apply the correspondence generator to generate putative correspondences.
+
+        Args:
+            client: Dask client, used to execute the front-end as futures.
+            images: List of all images, as futures.
+            image_pairs: Indices of the pairs of images to estimate two-view pose and correspondences.
+
+        Returns:
+            List of keypoints, one entry for each input images.
+            Putative correspondence as indices of keypoints, for pairs of images.
+        """
+        dataset_root = self._dataset_root
+        scene_name = self._scene_name
+
+        img_dir = f'{dataset_root}/{scene_name}'
+        poses_fpath = f'{dataset_root}/{scene_name}_COLMAP_SfM.log'
+        lidar_ply_fpath = f'{dataset_root}/{scene_name}.ply'
+        colmap_ply_fpath = f'{dataset_root}/{scene_name}_COLMAP.ply'
+        ply_alignment_fpath = f'{dataset_root}/{scene_name}_trans.txt'
+        bounding_polyhedron_json_fpath = f'{dataset_root}/{scene_name}.json'
+        loader = TanksAndTemplesLoader(
+            img_dir=img_dir,
+            poses_fpath=poses_fpath,
+            lidar_ply_fpath=lidar_ply_fpath,
+            ply_alignment_fpath=ply_alignment_fpath,
+            bounding_polyhedron_json_fpath=bounding_polyhedron_json_fpath,
+            colmap_ply_fpath=colmap_ply_fpath,
+        )
+
+        
+        mesh = loader.reconstruct_mesh()
+        # TODO(jolambert): File Open3d bug to add pickle support for TriangleMesh.
+        open3d_mesh_path = tempfile.NamedTemporaryFile(suffix='.obj').name
+        open3d.io.write_triangle_mesh(filename=open3d_mesh_path, mesh=mesh)
+
+        loader_future = client.scatter(loader, broadcast=False)
```

Comment:
We could just avoid futures for this file I guess?

Reply:
Hmm wouldn't we lose all parallelization benefits then?

Reply:
Yes, but since we are just loading from disk, we would not lose much in terms of time I think